lib/src/components/base_component.dart

```diff
@@ -181,14 +181,14 @@ abstract class BaseComponent extends Component {
   ) {
     bool shouldContinue = true;
     for (Component child in _children) {
-      if (child is T && child is BaseComponent) {
+      if (child is BaseComponent) {
         shouldContinue = child.propagateToChildren(handler);
-        if (shouldContinue) {
-          shouldContinue = handler(child);
-        }
-        if (!shouldContinue) {
-          break;
-        }
+      }
+      if (shouldContinue && child is T) {
+        shouldContinue = handler(child);
+      }
+      if (!shouldContinue) {
+        break;
       }
```

Comment:
this is unrelated but I think it was just wrong before

Reply:
Aah true, so that it doesn't break the tree if there is an unrelated component in between.